File: Kattis/halloweemirrors/halloweemirrors.py
from __future__ import division, print_function
range = xrange
input = raw_input
from itertools import izip as zip, imap as map, ifilter as filter

import math 

class Point:
    def __init__(self, x, y):
        self.x = x
        self.y = y

    def __str__(self):
        return "({}; {})".format(self.x, self.y)

    def __add__(self, other):
        return Point(self.x+other.x, self.y+other.y)

    def __sub__(self, other):
        return Point(self.x-other.x, self.y-other.y)

    def __mul__(self, c):
        return Point(self.x*c, self.y*c)

    def length_squared(self):
        return self.x**2 + self.y**2

    def length(self):
        return math.sqrt(self.length_squared())

    def cross(self, other):
        return self.x*other.y - self.y*other.x

    def dot(self, other):
        return self.x * other.x + self.y * other.y

    def angle(self, other):
        a = self.dot(other) / (self.length() * other.length())
        a = max(min(a, 1),-1)
        return math.acos(a)

n = int(input())

while n != 0:
    mirrors = []
    for _ in range(n):
        coords = list(map(int, input().split()))
        mirrors.append((Point(*coords[0:2]), Point(*coords[2:4])))

    m = int(input())
    for _ in range(m):
        loc = Point(*map(int, input().split()))
        count = 0
        for mirror in mirrors:
            a = mirror[1]-mirror[0]
            b = loc - mirror[0]

            c = mirror[0]-mirror[1]
            d = loc - mirror[1]

            if a.angle(b) <= math.pi/2 and c.angle(d) <= math.pi/2:
                count += 1
        print(count)
    print()


    n = int(input())

File: Kattis/halloweemirrors/halloweemirrors.cpp
#include <algorithm>
#include <utility>
#include <iostream>
#include <cmath>
#include <vector>

#define clamp(x, lo, hi) std::min(hi, std::max(lo, x))

template <typename T>
struct Point {
    T x, y;

    Point operator+(const Point & other) const {
        return {x+other.x, y+other.y};
    }

    Point operator-(const Point & other) const {
        return {x-other.x, y-other.y};
    }

    Point operator*(const T & k) const {
        return {x * k, y * k};
    }

    T length_squared() const {
        return x*x + y*y;
    }

    double length() const {
        return sqrt(length_squared());
    }

    T cross(const Point & other) const {
        return x*other.y - y*other.x;
    }

    T dot(const Point & other) const {
        return x*other.x + y*other.y;
    }

    double angle(const Point & other) const {
        double a = (dot(other) / length()) / other.length();
        return acos(clamp(a, -1.0, 1.0));
    }
};

template <typename T>
std::ostream & operator<<(std::ostream & out, const Point<T> & p) {
    return out << "(" << p.x << "; " << p.y << ")";
}

template <typename T>
std::istream & operator>>(std::istream & in, Point<T> & p) {
    return in >> p.x >> p.y;
}

typedef Point<int> P;

int main() {
    int n = 0; std::cin >> n;

    while (n != 0) {
        std::vector<std::pair<P, P>> mirrors(n);
        for (size_t i = 0; i < n; ++i) {
            std::cin >> mirrors[i].first >> mirrors[i].second;
        }

        int m; std::cin >> m; 
        for (size_t i = 0; i < m; ++i) {
            P loc; std::cin >> loc;
            size_t count = 0;
            for (auto mirror : mirrors) {
                auto a = mirror.second - mirror.first;
                auto b = loc - mirror.first;

                auto c = mirror.first - mirror.second;
                auto d = loc - mirror.second;

                // std::cerr << loc << " " << mirror.first << " " << mirror.second << std::endl;
                // std::cerr << a.angle(b) * 180/M_PI << " " << c.angle(d) * 180/M_PI << std::endl;
                if (a.angle(b) <= M_PI/2 && c.angle(d) <= M_PI/2) ++count;
            }

            std::cout << count << std::endl;
        }

        std::cout << std::endl;
        std::cin >> n;
    }
    return 0;
}
